janus_core/mlip_calculators.py

```diff
@@ -0,0 +1,101 @@
+"""Configure MLIP calculators.
+
+Similar in spirit with matcalc and quacc approaches
+- https://github.com/materialsvirtuallab/matcalc
+- https://github.com/Quantum-Accelerators/quacc.git
+"""
+
+from __future__ import annotations
+
+from typing import Literal
+
+from ase.calculators.calculator import Calculator
+
+architectures = ["mace", "mace_mp", "mace_off", "m3gnet", "chgnet"]
+
+
+def choose_calculator(
+    architecture: Literal[architectures] = "mace", **kwargs
+) -> Calculator:
+    """Choose MLIP calculator to configure.
+
+    Parameters
+    ----------
+    architecture : Literal[architectures], optional
+        MLIP architecture. Default is "mace".
+
+    Raises
+    ------
+    ModuleNotFoundError
+        MLIP module not correctly been installed.
+    ValueError
+        Invalid architecture specified.
+
+    Returns
+    -------
+    calculator : Calculator
+        Configured MLIP calculator.
+    """
+    # pylint: disable=import-outside-toplevel
+    # pylint: disable=too-many-branches
+    # pylint: disable=import-error
+    # Optional imports handled via `architecture`. We could catch these,
+    # but the error message is clear if imports are missing.
+    if architecture == "mace":
+        from mace import __version__
+        from mace.calculators import MACECalculator
+
+        if "default_dtype" not in kwargs:
+            kwargs["default_dtype"] = "float64"
```

Comment:
[`dict.setdefault`](https://docs.python.org/3/library/stdtypes.html#dict.setdefault) is there to handle this type of thing
```
kwargs.setdefault("default_dtype", "float64")
```